File: KidsWithTheGreatestNumberOfCandies.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

class Solution{
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies){
        List<Boolean> result = new ArrayList<>();
        int max = Arrays.stream(candies).max().getAsInt();
        for(int i = 0; i < candies.length; i++){
            if((candies[i] + extraCandies) >= max){
                result.add(true);
            }else{
                result.add(false);
            }
        }
        return result;
    }
    public static void main(String[] args){
        Solution test = new Solution();
        int[] arrTest = {4,2,1,1,2};
        int extraCand = 2;
        System.out.println(test.kidsWithCandies(arrTest, extraCand));
    }
}

File: kidsWiththeGreatestNumberOfCandies.js
/**
 * @param {number[]} candies
 * @param {number} extraCandies
 * @return {boolean[]}
 */
var kidsWithCandies = function(candies, extraCandies) {
    // let result = new Array();
    const max = Math.max(...candies);
    // candies.forEach((candie, index) => {
    //     if(candie + extraCandies >= max){
    //         result[index] = true;
    //     } 
    //     else result[index] = false;
    // });
    return candies.map((candie) => (candie + extraCandies) >= max);
};